app/components/multiple-textbox-input.js

```diff
@@ -3,16 +3,22 @@ import { computed } from '@ember/object';
 import Analytics from 'ember-osf/mixins/analytics';
 
 export default Component.extend(Analytics, {
-    legend: 'Text Fields',
-    textFields: null, // passed in array of text fields on invocation, default to null
+    model: null,
+    valuePath: null,
+    textFields: null,
     textFieldsLastIndex: computed('textFields.[]', function() {
         return this.get('textFields').length - 1;
     }),
 
     didReceiveAttrs() {
-        if (!this.textFields) {
-            this.set('textFields', [{ value: '' }]);
+        const valuesFromModel = this.model.get(this.valuePath);
+        if (valuesFromModel) {
+            this.set('textFields', valuesFromModel.map((x) => {
```

Comment:
shouldn't `valuesFromModel` be an array of links, why is the `valuesFromModel.map(...)` necessary?

Reply:
If you look at the function we pass to `map()`, you'll see that it returns `{ value: x }`. So what is does is transforming an array `['link1', 'link2', 'link3']` into another array `[{ value: 'link1'}, { value: 'link2'}, { value: 'link3'}]. The reason behind such transformation is because if we want the `value` binding of the input helper to work, we have to bind to objects instead of primitives.

Reply:
ah, right! Got things mixed up. 🤕